### PR TITLE
Add simulation for single component spin up

### LINKED ISSUE #29: Add code for simulating spin up of a viscous fluid
I have some code locally that simulates the response of a sphere of viscous fluid to s spin up on the boundary. This should be added to gains, as a basis for simulating the response of a coupled fluid/superfluid in the future.

### PR DESCRIPTION
**Pull request by Rmc2564**
Fixes #29 by uploading the code found in [this repo](https://github.com/Rmc2564/Ekman_vc) and refactoring it to be used in a package. The content added can be summarised as follows:

`scripts/single_spin_up_rotating_frame.py` uses Dedalus to simulate a spherical fluid response to a spin up at the equator, and saves snapshots to the `outputs` directory. The simulation parameters are read the `src/gains/params/single_spin_up_rotating.py` file.

`src/gains/initial_conditions/single_component_spin_up.py` contains the `window()` function. This is close to 1 in a specified width and close to 0 everywhere else, and is used to enforce boundary conditions at specific regions on the sphere rather than the whole surface.

`tests/test_initial_conditions/test_single_spin_up.py` Contains unit tests for the `window` function.

`src/gains/Analysis/analyse_spin_up` Contains various functions used in plotting and analysing the results of the spin up simulation. These functions were moved out of scripts as they may be useful/adaptable to analyse future simulations.

`scripts/plot_spin_up.py` Plots the results of the spin up simulation and saves the figures to `outputs`. An optional "frames" directory is also added to generate plots of all (around 500) snapshots for creating animations.

Excluded codes `G004` and `PTH208` from the ruff hook when linting

### CODE REVIEW

File: scripts/single_spin_up_rotating_frame.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+# Solver
+solver = problem.build_solver(timestepper)
+solver.stop_sim_time = PARAMS["stop_sim_time"]
+
+use_checkpoint = False
+
+if use_checkpoint:
+    write, timestep = solver.load_state(
+        "outputs/su_equator/checkpoints/checkpoints_sNumber.h5"
```
Comment (ilectra) [inline scripts/single_spin_up_rotating_frame.py]: Take the output directory from command line, like you did in #25 . Then the path to sub-directories here and elsewhere in the script will be relative to that.
Or, in addition, you can define a hierarchy of directories in the PARAMS structure - if it makes sense.

File: src/gains/Analysis/analyse_spin_up.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        self.coord = coord
+        self.label = label
+
+    def getcoord(self) -> np.ndarray:
+        """Get the coordinate."""
+        return self.coord
+
+    def getlabel(self) -> str:
+        """Get the label."""
+        return self.label
```
Comment (willGraham01) [inline src/gains/Analysis/analyse_spin_up.py]: Just a nitpick here - these functions don't actually provide any functionality over just attribute access. IE doing `LabeledCoordinate.getccord()` would be functionally identical to the direct access of `LabeledCoordinate.coord`. This is a Python thing which I can elaborate more on if you want me to - it differs from languages like C/C++ where the concept of private/public/friend attributes is actually enforced.

`coord` and `label` are not private class attributes (and in Python such things don't really exist anyway), so you don't need a "getter" method for them. Further to this point, you're returning the underlying `np.ndarray`s anyway, which will be updated in place if they are manipulated so this isn't a "pure getter" anyway.

TLDR Python doesn't believe in privacy. Don't bother with `get<thing>` here and just access the underlying attribute.

File: pyproject.toml [hunk 3]
```diff
@@ -60,6 +61,8 @@ lint.ignore = [
   "D407",   # removed dashes lines under sections
   "D417",   # argument description in docstring (unreliable)
   "ISC001", # simplify implicit str concatenation (ruff-format recommended)
+  "G004", # Logging statement uses f-string (Conflicts with a fix)
+  "PTH208", # Forces use of pathlib instead of os.listdir
```
[the remark this answers is not in the document]
Comment (Rmc2564) [reply pyproject.toml]: There probably was when I excluded it initially, but the linting passes fine if I remove the line excluding it so I have done that.
